### PR TITLE
add workflow to run tests

### PR DESCRIPTION
**Pull request by wilsoncusack**

### CODE REVIEW

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            echo "First 5 characters: ${VITE_ANVIL_FORK_URL:0:15}"
+          fi
+        env:
+          VITE_ANVIL_FORK_URL: ${{ secrets.VITE_ANVIL_FORK_URL }}
+          
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up pnpm
+        uses: pnpm/action-setup@v2
```
Comment (roninjin10) [inline .github/workflows/verify.yml:32]: I thought this repo used yarn?

Comment (roninjin10) [reply .github/workflows/verify.yml:32]: oh i see you migrated to pnpm in this pr

Comment (wilsoncusack) [reply .github/workflows/verify.yml:35]: haha yeah, @roninjin10 what do you think? Thought maybe worth it for slight speedup

**Comment by wilsoncusack** [comment]
> Feels early to start adding tests tbh. Usually would recommend devs either do one of the following first:
> 
> 1. Write a spec for the api first they feel confident about
> 2. Wait til they have implemented at least the difficult parts of the api (in this case withdrawals) and feel confident they have a stable api
>    For some devs tests are a part of their workflow because they do TDD or run tests in watch mode to do development and in this case it's great but for most devs on green projects tests are something that slows them down

Thanks, yeah I prefer more test driven development and sort of "do it right the first time." I think it helps ensure things are actually working, and promises to circle back and add coverage often don't workout in my experience. I think we could spike on a doc if we wanted to try to nail down the API a bit more, but I am also OK circling back and updating tests.

**Comment by roninjin10** [comment]
> promises to circle back and add coverage often don't workout in my experience
I have tricks for this that I think is actual useful in general! Worked for me at startups and at giant corporations. Just sharing because this is one of my favorite engineering related things to talk about.

1. Never ship unproductionized code. Have clear guidelines on what productionized means. Don't promise to circle back always make the productionization (tests, metrics, clean code conventions etc.) part of the product requirement.
2. Ship prs/changes very fast. Teams and individuals fail to circle back and add coverage because time is the bottleneck. This means you gotta make writing them easy, and you have to increase velocity of shipping prs and changes in general.
3. Related to private point I advocate a rule that the moment code has improved the codebase more than before it should land. I'd rather see 20 iterative prs to a goal than 5 perfect prs.

**Comment by roninjin10** [comment]
Another interesting thing I love to talk about is how tests is tests are different for libraries and applications. For applications I always tell folks that tests don't exist for quality reasons. If you want quality, hire a team of 20 people to manually test your code before every release. That's a higher quality bar than tests.  Tests in applications exist to make your life easier and ship faster. If developers feel like tests are a time consuming chore that means either the tests they are writing aren't actually valuable or they aren't thinking long term (maybe they are a contractor who won't have to maintain the code). By definition good tests should be saving time.

Library code is slightly different. I tend to aim for near 100% test coverage on pure library code and I think library code in general should actually use tests to guarantee quality unlike application code

**Comment by roninjin10** [comment]
Mentioned the library code to application code because the same thing applies here. The api being implemented right now isn't even what I would ship as the final API if I end up bringing this project to completion. In that way this is more like "application code" than "library code" at this time. Nailing down the API is what turns it into library code in terms of how the tradeoffs work out but nailing down the perfect API is also a lot of work compared to just figuring it out iteratively

**Pull request merged**